### ISSUE TITLE
Adding weight maps to binary_crossentropy

### ISSUE THREAD

**Issue by Shankarram2709** [jaidevd/jaidevd.github.io#4]
Incase single class segmentation where I am concentrating on certain areas in masks to have higher weights how do we add up the weights.

I created a weight map as shape as mask (256,256,1).
As you suggested computed the loss as a part of computational graph.
```
final_layer=layer.Lambda(weightedloss,output_shape=(256,256,1))([output,weightmap,mask])
def weightedloss(x):
  output,weightmap,mask=x
  loss=K.binary_crossentropy(output,mask)
  loss=layers.multiply(loss*weightmap)
  return loss

model(inputs=[inputs,weightmap,mask],output=loss)
```
I developed this based on the this [answer.](https://stackoverflow.com/questions/48555820/keras-binary-segmentation-add-weight-to-loss-function/62214983#62214983)

Eventhough it seems to work well my Validation metric of IOU goes higher than 1.0 even more than 5.0 at times.
And also K.binary_crossentropy(output,mask) I think this should be the other way around(mask,output) and when I tried IOU decreased. 

Also I got to know that binary crossentropy is computed only with logits. 

Do you have a solution for also binarycrossentropy loss?
